src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/config/WebClientConfiguration.kt

```diff
@@ -34,4 +41,31 @@ class WebClientConfiguration(
   @Bean
   fun mappingApiWebClient(authorizedClientManager: ReactiveOAuth2AuthorizedClientManager, builder: WebClient.Builder): WebClient =
     builder.reactiveAuthorisedWebClient(authorizedClientManager, registrationId = "nomis-mapping-api", url = mappingApiBaseUri, timeout)
+
+  @Bean
+  fun reactiveOAuth2AuthorizedClientManagerWithTimeout(
+    reactiveClientRegistrationRepository: ReactiveClientRegistrationRepository,
+    reactiveOAuth2AuthorizedClientService: ReactiveOAuth2AuthorizedClientService,
+  ): ReactiveOAuth2AuthorizedClientManager = AuthorizedClientServiceReactiveOAuth2AuthorizedClientManager(
+    reactiveClientRegistrationRepository,
+    reactiveOAuth2AuthorizedClientService,
+  ).apply {
+    val accessTokenResponseClient =
```

Comment:
this really the only difference to the bean in the library, manually add a client with a timeout, the standard code just calls WebClient.builder().build()

Reply:
looks a lot of code to have to write to override, I'll have a play and see if there is a simpler way

Reply:
the simpler way is to add the code to the library instead 😂

Reply:
100% - but want to test it first.

BTW I'm not sure if it is as easy as you would imagine @petergphillips 

The bean creation here has access to the required timeout value, the bean creating in the library is separated from the main webClient creation that knows about the timeout.

I think I will leave it for @mikehalmamoj  - since he will probably want some fancy way of coinfiguring the auth timeout for each webClient where as my solution sets each web client auth client to have the same timeout since that is exactly what we want. Others my have other requirements.